src/components/ShareSaveQuery/SaveQuery.js

```diff
@@ -73,163 +78,162 @@ export default injectState(
       const { disabled } = this.props;
       return (
         !!this.props.state.loggedInUser && (
-          <ButtonContainer>
-            <WhiteButton
-              disabled={disabled}
-              onClick={() =>
-                !disabled &&
-                this.setState({ open: true }, () => {
-                  // so hacky, but couldn't get any reasonable approach to work
-                  let id = setInterval(() => {
-                    if (this.input) {
-                      this.input.focus();
-                      clearInterval(id);
-                    }
-                  });
-                })
-              }
-            >
-              <Route>
-                {({ history }) => (
-                  <Tooltip
-                    position="bottom"
-                    onRequestClose={() => {
-                      this.setState({ open: false, link: null });
-                    }}
-                    interactive
-                    open={this.state.open}
-                    html={
-                      <Component
-                        initialState={{ saved: false }}
-                        render={({ state, setState }) => (
+          <WhiteButton
+            disabled={disabled}
+            onClick={() =>
+              !disabled &&
+              this.setState({ open: true }, () => {
+                // so hacky, but couldn't get any reasonable approach to work
+                let id = setInterval(() => {
```

Comment:
At least, add a timestamp before, and stop if the interval is running since more than enough time, to avoid memory leaks.

Reply:
great point :) although this was actually unmodified existing code, I just removed the wrapping `<ButtonContainer/>` to support layout :P